### ISSUE TITLE
`jax.nn.sigmoid` leads to `nan` when calculating gradient if input is large

### ISSUE THREAD

**Issue by BirkhoffG** [BirkhoffG/cfnet#11]
Reference:
https://stackoverflow.com/a/68293931
